src/main/java/net/imglib2/parallel/Parallelization.java

```diff
@@ -0,0 +1,266 @@
+	 */
+	public static < R > R runWithNumThreads( int numThreads, Callable< R > action )
+	{
+		try (TaskExecutor taskExecutor = TaskExecutors.numThreads( numThreads ))
+		{
+			return runWithExecutor( taskExecutor, action );
+		}
+	}
+
+	/**
+	 * Executes the given {@link Runnable} with the given {@link ExecutorService},
+	 * and waits for the execution to finish.
+	 */
+	public static void runWithExecutor( ExecutorService executorService, Runnable action )
+	{
+		runWithExecutor( TaskExecutors.forExecutorService( executorService ), action );
+	}
+
+	/**
+	 * Executes the given {@link Callable} with the given {@link ExecutorService},
+	 * waits for the execution to finish and returns the result.
+	 */
+	public static < R > R runWithExecutor( ExecutorService executorService, Callable< R > action )
+	{
+		return runWithExecutor( TaskExecutors.forExecutorService( executorService ), action );
+	}
+
+	/**
+	 * Executes the given {@link Runnable} with the given {@link TaskExecutor},
+	 * and waits for the execution to finish.
+	 */
+	public static void runWithExecutor( TaskExecutor taskExecutor, Runnable action )
+	{
+		try (Frame frame = setExecutorRequiresReset( taskExecutor ))
+		{
+			action.run();
+		}
+	}
+
+	/**
+	 * Executes the given {@link Callable} with the given {@link TaskExecutor},
+	 * waits for the execution to finish and returns the result.
+	 */
+	public static < T > T runWithExecutor( TaskExecutor taskExecutor, Callable< T > action )
+	{
+		try (Frame frame = setExecutorRequiresReset( taskExecutor ))
+		{
+			return action.call();
+		}
+		catch ( Exception e )
+		{
+			throw new RuntimeException( e );
+		}
+	}
+
+	/**
+	 * This method can be used to execute an algorithm with a given {@link TaskExecutor}.
+	 * But it's easier to use {@link #runWithExecutor}.
+	 * <p>
+	 * This method sets the {@link TaskExecutor} for the current thread.
+	 * This can be used to execute a certain part of your code with the given
+	 * {@link TaskExecutor}. It's mandatory to call the {@code close()} of the
+	 * return {@link Frame frame} afterwards. This could be done using an
+	 * try-with-resources statement, that was introduced with Java 7.
+	 * <pre>
+	 * {@code
+	 *
+	 * try ( Parallelization.Frame frame = Parallelization.setExecutorRequiresReset( taskExecutor ) )
+	 * {
+	 *     myAlgorithm(input);
+	 * }
+	 * }
+	 * </pre>
+	 * Or by explicitly calling {@code frame.close()} in the finally block.
+	 * <pre>
+	 * {@code
+	 *
+	 * Parallelization.Frame frame = Parallelization.setExecutorRequiresReset( taskExecutor );
+	 * try
+	 * {
+	 *     myAlgorithm(input);
+	 * }
+	 * finally
+	 * {
+	 *     frame.close();
+	 * }
+	 * }
+	 * </pre>
+	 */
+	public static Frame setExecutorRequiresReset( TaskExecutor taskExecutor )
+	{
+		final TaskExecutor old = executor.get();
+		executor.set( taskExecutor );
+		return () -> executor.set( old );
+	}
+
+	public interface Frame extends AutoCloseable
```

Comment:
It's confusing that Frame has a `void close()` and `void reset()` method that do the same thing.
Why not simply use `close()`? (You even suggest to use close() in the `setExecutorRequiresReset` javadoc)
I.e., this could be simply `public interface Frame extends AutoCloseable {}`.

Reply:
Done. I removed the reset method.